pandas/core/dtypes/cast.py

```diff
@@ -1018,6 +1020,80 @@ def soft_convert_objects(
     return values
 
 
+def convert_dtypes(
+    input_array,
+    convert_string: bool = True,
+    convert_integer: bool = True,
+    convert_boolean: bool = True,
+) -> Dtype:
+    """
+    Convert objects to best possible type, and optionally,
+    to types supporting ``pd.NA``.
+
+    Parameters
+    ----------
+    input_array : ExtensionArray or PandasArray
```

Comment:
"ExtensionArray or PandasArray" is redundant, isnt it?  is ndarray not allowed?  either way, can input_array be annotated?

Reply:
@jbrockmendel You're correct about the redundancy (this description resulted after lots of discussion above), and I think an ndarray would work, but it is probably untested. 

With respect to annotation, the issue here is the ordering of imports, so if it were to be typed, it requires changes to `_typing.py` and I didn't want to introduce that complexity to the PR.

Reply:
thanks for explaining, my mistake not following the thread in real-time.